static/app/views/dashboardsV2/gridLayout/widgetCard.tsx

```diff
@@ -0,0 +1,328 @@
+
+import {openDashboardWidgetQuerySelectorModal} from 'app/actionCreators/modal';
+import {Client} from 'app/api';
+import {HeaderTitle} from 'app/components/charts/styles';
+import ErrorBoundary from 'app/components/errorBoundary';
+import FeatureBadge from 'app/components/featureBadge';
+import MenuItem from 'app/components/menuItem';
+import {isSelectionEqual} from 'app/components/organizations/globalSelectionHeader/utils';
+import {Panel} from 'app/components/panels';
+import Placeholder from 'app/components/placeholder';
+import {IconDelete, IconEdit, IconGrabbable} from 'app/icons';
+import {t} from 'app/locale';
+import overflowEllipsis from 'app/styles/overflowEllipsis';
+import space from 'app/styles/space';
+import {GlobalSelection, Organization} from 'app/types';
+import trackAdvancedAnalyticsEvent from 'app/utils/analytics/trackAdvancedAnalyticsEvent';
+import {DisplayModes} from 'app/utils/discover/types';
+import withApi from 'app/utils/withApi';
+import withGlobalSelection from 'app/utils/withGlobalSelection';
+import withOrganization from 'app/utils/withOrganization';
+import ContextMenu from 'app/views/dashboardsV2/contextMenu';
+import {Widget} from 'app/views/dashboardsV2/types';
+import {eventViewFromWidget} from 'app/views/dashboardsV2/utils';
+import {DisplayType} from 'app/views/dashboardsV2/widget/utils';
+import WidgetCardChart from 'app/views/dashboardsV2/widgetCardChart';
+import WidgetQueries from 'app/views/dashboardsV2/widgetQueries';
+
+export const DRAG_HANDLE_ID = 'grid-layout-drag-handle';
+
+type Props = WithRouterProps & {
+  api: Client;
+  organization: Organization;
+  location: Location;
+  isEditing: boolean;
+  widget: Widget;
+  selection: GlobalSelection;
+  onDelete: () => void;
+  onEdit: () => void;
+  isSorting: boolean;
+  currentWidgetDragging: boolean;
+  showContextMenu?: boolean;
+  hideToolbar?: boolean;
+  renderErrorMessage?: (errorMessage?: string) => React.ReactNode;
+  noLazyLoad?: boolean;
+};
+
+export class WidgetCard extends React.Component<Props> {
+  shouldComponentUpdate(nextProps: Props): boolean {
+    if (
+      !isEqual(nextProps.widget, this.props.widget) ||
+      !isSelectionEqual(nextProps.selection, this.props.selection) ||
+      this.props.isEditing !== nextProps.isEditing ||
+      this.props.isSorting !== nextProps.isSorting ||
+      this.props.hideToolbar !== nextProps.hideToolbar
+    ) {
+      return true;
+    }
+    return false;
+  }
+
+  isAllowWidgetsToDiscover() {
+    const {organization} = this.props;
+    return organization.features.includes('connect-discover-and-dashboards');
+  }
+
+  renderToolbar() {
+    const {onEdit, onDelete, hideToolbar, isEditing} = this.props;
+
+    if (!isEditing) {
+      return null;
+    }
+
+    return (
+      <ToolbarPanel>
+        <IconContainer style={{visibility: hideToolbar ? 'hidden' : 'visible'}}>
+          <IconClick>
+            <StyledIconGrabbable id={DRAG_HANDLE_ID} color="textColor" />
```

Comment:
Don't think this causes any issues right now, but would it possible to use a different selector? This will produce multiple elements with same ids which we probably want to avoid.

Reply:
RGL just requires a CSS selector. Would a `className` be fine? I'm thinking of calling it `DRAG_HANDLE_CLASS`, let me know if you disagree with the name

Reply:
I think `className` is probably fine!